pkg/controller/master/supportbundle/controller.go

```diff
@@ -50,6 +67,66 @@ func (h *Handler) OnSupportBundleChanged(key string, sb *harvesterv1.SupportBund
 	}
 }
 
+func (h *Handler) getSupportBundleImage() (string, corev1.PullPolicy, error) {
+	harvesterManagedChart, err := h.managedChartCache.Get(settingctl.ManagedChartNamespace, settingctl.HarvesterManagedChartName)
+	if err != nil {
+		return "", corev1.PullPolicy(""), err
+	}
+
+	harvesterApp, err := h.appCache.Get(harvesterManagedChart.Spec.DefaultNamespace, harvesterManagedChart.Spec.ReleaseName)
+	if err != nil {
+		return "", corev1.PullPolicy(""), err
+	}
+
+	values, err := chartutil.CoalesceValues(
+		&chart.Chart{
+			// In latest version of chartutil, it read chart.metadata.name, so we put a value here.
+			// ref: https://github.com/helm/helm/blob/b8d3535991dd5089d58bc88c46a5ffe2721ae830/pkg/chartutil/coalesce.go#L160
+			Metadata: &chart.Metadata{Name: "merge-templates-and-values"},
+			Values:   harvesterApp.Spec.Chart.Values,
+		},
+		harvesterApp.Spec.Values,
+	)
+	if err != nil {
+		return "", corev1.PullPolicy(""), fmt.Errorf("can't merge yaml template and values in apps %s/%s", harvesterManagedChart.Spec.DefaultNamespace, harvesterManagedChart.Spec.ReleaseName)
+	}
+
+	var supportBundleYaml map[string]interface{}
+	if v, ok := values[SupportBundleRespository]; !ok {
+		return "", corev1.PullPolicy(""), fmt.Errorf("cant't find %s in apps %s/%s", SupportBundleRespository, harvesterManagedChart.Spec.DefaultNamespace, harvesterManagedChart.Spec.ReleaseName)
+	} else {
+		if supportBundleYaml, ok = v.(map[string]interface{}); !ok {
+			return "", corev1.PullPolicy(""), fmt.Errorf("unknown %s yaml struct %+v in apps %s/%s", SupportBundleRespository, v, harvesterManagedChart.Spec.DefaultNamespace, harvesterManagedChart.Spec.ReleaseName)
+		}
+	}
+	return h.getSupportBundleImageFromYaml(supportBundleYaml)
+}
+
+func (h *Handler) getSupportBundleImageFromYaml(supportBundleYaml map[string]interface{}) (string, corev1.PullPolicy, error) {
+	// support bundle yaml struct
+	// support-bundle-kit:
+	//   image:
+	//     imagePullPolicy: <string>
+	//     repository: <string>
+	//     tag: <string>
+	var ok bool
+	var imageYaml map[string]interface{}
+
+	if _, ok = supportBundleYaml["image"]; !ok {
+		return "", corev1.PullPolicy(""), fmt.Errorf("image is not in %s yaml", SupportBundleRespository)
+	}
+	if imageYaml, ok = supportBundleYaml["image"].(map[string]interface{}); !ok {
+		return "", corev1.PullPolicy(""), fmt.Errorf("unknown image yaml struct in %s", SupportBundleRespository)
+	}
+
+	for _, key := range []string{"repository", "tag", "imagePullPolicy"} {
+		if _, ok = imageYaml[key]; !ok {
+			return "", corev1.PullPolicy(""), fmt.Errorf("%s is not in %s.image yaml", key, SupportBundleRespository)
+		}
+	}
+	return fmt.Sprintf("%s:%s", imageYaml["repository"].(string), imageYaml["tag"].(string)), corev1.PullPolicy(imageYaml["imagePullPolicy"].(string)), nil
+}
```

Comment:
[`mapstructure`](https://pkg.go.dev/github.com/mitchellh/mapstructure) is in harvester's dependency graph, so I guess we can just use it and it's really self-explaining. Something like this [go playground](https://go.dev/play/p/uHkrVGfhnw1)

```go
func (h *Handler) getSupportBundleImageFromYaml(supportBundleYaml map[string]interface{}) (string, corev1.PullPolicy, error) {
	// support bundle yaml struct
    var imageYaml struct {
		Image struct {
			ImagePullPolicy corev1.PullPolicy
			Repository string
			Tag string
		}
	}
	if err := mapstructure.Decode(supportBundleYaml, &imageYaml); err != nil {
		return "", corev1.PullPolicy(""), err
	}
	return fmt.Sprintf("%s:%s", imageYaml.Image.Repository, imageYaml.Image.Tag), imageYaml.Image.ImagePullPolicy, nil
}
```

Reply:
Thanks for your suggestion! I move `getSupportBundleImageFromYaml` to `util.GetImageFromYaml`.